api/include/opentelemetry/trace/span.h

```diff
@@ -1,38 +1,112 @@
 #pragma once
 
+#include <cstdint>
+
 #include "opentelemetry/core/timestamp.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/trace/canonical_code.h"
 
 namespace opentelemetry
 {
 namespace trace
 {
+enum class SpanKind
```

Comment:
Do we want `kUnspecified`? Would this be int or uint (or uint8)?

https://github.com/open-telemetry/opentelemetry-proto/blob/2646c0b87fa6059a9ee85e12e6d0007ff794d829/opentelemetry/proto/trace/v1/trace.proto#L93

Reply:
otel-java has `INTERNAL` as a default value.

Reply:
IMO let's leave it as-is to match what Java does: https://github.com/open-telemetry/opentelemetry-java/blob/42bff3e9223181af800c76186744472c13499b0a/api/src/main/java/io/opentelemetry/trace/Span.java#L41